selfhost/typechecker.jakt

```diff
@@ -6932,144 +6932,118 @@ struct Typechecker {
                     expr: range
                     call: ParsedCall(name: "iterator")
                     is_optional: false
-                    span: name_span
+                    span: range.span()
                 )
             }
         } else {
             resolved_iterable_result_type = iterable_trait_implementation!.implemented_type_args[0]
         }
 
```

Comment:
diff got messy here. I:
- removed the comments above the rewrite
- turned rewritten statement from `ParsedStatement::Block` into `ParsedBlock`
- patched a couple of spans so that ide resolves correctly.